pkg/controller/identity/iampolicy/controller.go

```diff
@@ -275,3 +282,44 @@ func (e *external) deleteNonDefaultVersions(ctx context.Context, policyArn strin
 
 	return nil
 }
+
+// ListPoliciesPages paginates ListPolicies
+func (e *external) ListPoliciesPages(ctx context.Context, input *awsiam.ListPoliciesInput, fn func(*awsiam.ListPoliciesOutput, bool) bool) error {
+	for {
+		output, err := e.client.ListPolicies(ctx, input)
+		if err != nil {
+			return err
+		}
+		lastPage := awsclient.StringValue(output.Marker) == ""
+		if !fn(output, lastPage) || lastPage {
+			break
+		}
+		input.Marker = output.Marker
+	}
+	return nil
+}
+
+// getPolicyArnByName will paginate through all policies to retrieve by name instead of arn
+func (e *external) getPolicyArnByName(ctx context.Context, policyName string) (*string, error) {
+
+	input := &awsiam.ListPoliciesInput{Scope: awsiamtypes.PolicyScopeTypeLocal}
+
+	policyArn := awsclient.String("")
```

Comment:
This is confusing to me, and it's compared against nil later. Is it simpler with this?
```suggestion
	var policyArn *string
```